node-gui/backend/src/messages.rs

```diff
@@ -62,7 +62,23 @@ impl AccountId {
     }
 }
 
-#[derive(Debug, Clone)]
+impl Serialize for AccountId {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        self.0.into_u32().serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for AccountId {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        let i = u32::deserialize(deserializer)?;
+        let i = U31::from_u32(i).ok_or(D::Error::custom(format!(
```

Comment:
1) `ok_or_else` to avoid constructing the error string when there is no error.

2) it's better to have a test for this.